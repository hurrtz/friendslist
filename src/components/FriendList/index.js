import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import FriendListItem from '../FriendListItem';

import styles from './styles.css';

class FriendList extends PureComponent {
  render() {
    const { friends, actions } = this.props;

    return (
      <ul className={styles.friendList}>
        {friends.map((friend, index) => {
          return (
            <FriendListItem
              key={index}
              id={index}
              name={friend.name}
              starred={friend.starred}
              {...actions}
            />
          );
        })}
      </ul>
    );
  }
}

FriendList.propTypes = {
  friends: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired,
};

export default FriendList;
