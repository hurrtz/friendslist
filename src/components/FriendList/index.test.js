import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import FriendList from './';

Enzyme.configure({ adapter: new Adapter() });

function setup(injectedProps = {}) {
  const props = {
    friends: [{ name: 'Name1', starFriend: () => {}, setSex: () => {} }],
    actions: {},
    ...injectedProps,
  };

  const enzymeWrapper = shallow(<FriendList {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('FriendList', () => {
  it('should render self and subcomponents', () => {
    const { enzymeWrapper, props } = setup({
      friends: [{ name: 'Name1' }],
      actions: { starFriend: () => {}, setSex: () => {} },
    });

    expect(enzymeWrapper.find('ul').exists()).toBe(true);
    expect(enzymeWrapper.find('FriendListItem').exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<FriendList friends={[]} actions={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
