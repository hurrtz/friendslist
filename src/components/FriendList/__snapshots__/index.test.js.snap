exports[`FriendList should match the snapshot 1`] = `
<ul
  className={undefined} />
`;
